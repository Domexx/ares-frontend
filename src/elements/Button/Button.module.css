/* === === === === Button === === === === */

.Button {
  padding: 28px 20px;
  border-radius: 4px;
  font-size: 14px;
  font-weight: 500;
  line-height: 17px;
  text-decoration: none;
  outline: none;
  transition: filter 250ms ease-in-out;
}

.Button--default {
  background: var(--clr-clickable);
  color: #000000;
}

.Button--default:hover {
  filter: brightness(85%);
}

.Button--secondary {
  background: var(--clr-dark-600);
  color: #ffffff;
  transition: background 250ms ease-in-out;
}

.Button--secondary:hover {
  background: var(--clr-dark-500);
}
