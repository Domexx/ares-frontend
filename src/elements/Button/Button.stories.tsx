import { Story } from '@storybook/react/types-6-0';
import React, { ComponentProps } from 'react';
import ButtonComponent, { Props } from './Button';

const Component = {
  title: 'Elements/Button',
  component: ButtonComponent,
  argTypes: {},
};

const Template: Story<ComponentProps<typeof ButtonComponent>> = (args) => {
  return <ButtonComponent {...args} />;
};

export const Button = Template.bind({});
export default Component;

// https://storybook.js.org/docs/react/essentials/controls
Button.args = {
  children: 'Test',
} as Props;
