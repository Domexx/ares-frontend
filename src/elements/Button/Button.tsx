import clsx from 'clsx';
import React, { PropsWithChildren } from 'react';

import s from './Button.module.css';

type ButtonType = 'default' | 'primary' | 'secondary' | 'danger' | 'warning';

type ButtonAlign = 'left' | 'right' | 'center';

export type Props = {
  onClick?: () => void;
  disabled?: boolean;
  loading?: boolean;
  type?: ButtonType;
  align?: ButtonAlign;
  big?: boolean;
  small?: boolean;
  className?: string;
  href?: string;
};

const Button: React.FC<PropsWithChildren<Props>> = ({
  onClick,
  disabled,
  loading = true,
  type = 'default',
  align = 'left',
  big,
  small,
  className = '',
  children,
  ...rest
}) => {
  const handleClick = () => {
    if (typeof onClick === 'function' && !disabled && !loading) {
      onClick();
    }
  };

  const classNames = clsx(
    s['Button'],
    s[`Button--${type}`],
    s[`Button--${align}`],
    `${big ? s['Button--big'] : ''}`,
    `${small ? s['Button--small'] : ''}`,
    `${small ? s['Button--disabled'] : ''}`,
    className,
  );

  // Render functional Button (default)
  return (
    <button type="button" className={classNames} onClick={handleClick} {...rest}>
      {children}
    </button>
  );
};

export default Button;
