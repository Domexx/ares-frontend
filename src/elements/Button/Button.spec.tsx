import React from 'react';

import { render } from '@testing-library/react';

import Button from './Button';

describe('Button Component', () => {
  it('should render properly', () => {
    const { container } = render(<Button />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
