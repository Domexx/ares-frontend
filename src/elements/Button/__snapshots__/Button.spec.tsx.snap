// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should render properly 1`] = `
<button
  class="Button Button--default Button--left"
  type="submit"
/>
`;
